indico/modules/events/registration/client/js/form/FormItem.jsx

```diff
@@ -149,10 +165,15 @@ export default function FormItem({
       </>
     ) : null;
 
+  if (!show && !setupMode) {
+    form.change(rest.htmlName, null);
+    return null;
+  }
+
   return (
     <div
       styleName={`form-item ${toClasses({
-        'disabled': !isEnabled || paidItemLocked,
+        'disabled': !isEnabled || paidItemLocked || !show,
```

Comment:
> I changed it so that in the management section the "hidden" field is greyed to indicate it will not be displayed in the user's section.

I'd prefer to not use `disabled` styling but something custom. "Disabled" is not very visible (very low contrast), and I think not everyone realizes what is means. Maybe you can get an idea from other things like when a field is locked for having been already paid. It's very visible and has an icon with a tooltip explaining why it looks differently. IMHO something similar (maybe looking less "locked" and not so stripy) would be good here!

![image](https://github.com/user-attachments/assets/c205465e-7c08-40a1-98ad-9e4999f4dc8c)

Reply:
What about this?

<img width="835" alt="Screenshot 2025-01-16 at 12 36 57" src="https://github.com/user-attachments/assets/cac3eb02-313d-4529-8c02-42beb4311a88" />

Reply:
Better :) Just some random thoughts (no idea if they look good or not):

- different background color (`$pastel-blue` or `$light-purple` maybe)
- no stripes?

And of course an icon in the corner with a tooltip on hover explaining that the field is conditionally shown.

Bonus: you always show the icon, and just adjust the background depending on whether the condition is met or not. We do something similar (showing an icon) on fields if they have a retention period set.

Generally it's important that someone can immediately see on that page that a field has a condition, regardless of whether it's met or not. Otherwise maintaining bigger and more complex registration forms is a mess if you have to check individual config dialogs to figure this out.

Reply:
With pastel-blue:
<img width="802" alt="Screenshot 2025-01-16 at 12 56 33" src="https://github.com/user-attachments/assets/4f42e364-acf1-4f2a-9f00-c867fbc95a6d" />

With light-purple:
<img width="792" alt="Screenshot 2025-01-16 at 12 56 52" src="https://github.com/user-attachments/assets/56c4ae25-5adc-4493-bdd4-ddfe00cbf03f" />

Not a strong opinion at all, but my intuition tells me if something has stripes it would be hidden.

Reply:
pastel-blue + stripes maybe? light-purple is too dark imho

Reply:
<img width="791" alt="Screenshot 2025-01-16 at 13 09 43" src="https://github.com/user-attachments/assets/3837db29-4e1b-42a6-8bfa-061170361986" />

This is a combination of pastel-blue and light blue. What do you think?

Reply:
looks ok :)

Reply:
And with the always-visible icon:

<img width="1025" alt="Screenshot 2025-01-16 at 13 17 02" src="https://github.com/user-attachments/assets/7bb1eca6-e8fb-433c-ba67-0d6ff70c86e1" />